development-tools/clinic.rst

```diff
@@ -2001,6 +2001,43 @@ The generated glue code looks like this:
 .. versionadded:: 3.13
 
 
+.. _clinic-howto-getter:
+
+How to generate a getter
+------------------------
+
+You can use the ``@getter`` directive to generate an "impl" function
+to define a `getter <https://docs.python.org/3/c-api/structures.html#c.PyGetSetDef.get>`_.
+
+Example from :cpy-file:`Modules/_io/bufferedio.c`::
+
+    /*[clinic input]
+    @critical_section
+    @getter
+    _io._Buffered.closed
+    [clinic start generated code]*/
+
+The generated glue code looks like this.
+Note that this example is a mixture using the ``@critical_section`` directive
+to achieve thread safety without causing deadlocks between threads:
```

Comment:
To me this reads slightly weird; I expected the generated code snippet to follow straight after the first sentence! :) Also, to me the "note" sentence belongs _before_ the clinic input, not after it.

Moreover, is the `@critical_section` needed for the example? If not, I think we should remove it, so the example is stripped down to the bare minimum. OTOH, if all getters actually have a critical section (probably most of them do, @colesbury?), perhaps the `@getter` directive should imply `@critical_section`; it would reduce clinic input boiler plate code.

Reply:
> Moreover, is the @critical_section needed for the example?

If we attach the real example, "yes"

> perhaps the @getter directive should imply @critical_section; it would reduce clinic input boiler plate code.

I think that it should be optional if possible.

Reply:
Ok, so let's leave the `@critical_section` directive in the example for now. I still would like to mention that _before_ the clinic input, though. Also, we should link to the `@critical_section` how-to.